server/middleware/populateUserAuthorities.ts

```diff
@@ -0,0 +1,18 @@
+import { RequestHandler } from 'express'
+import jwtDecode from 'jwt-decode'
+import { ApplicationRoles } from './roleBasedAccessControl'
+
+export default function populateUserAuthorities(): RequestHandler {
```

Comment:
A simple middleware function that creates 2 variables on response locals: `hasEditAuthority` and `hasViewAuthority` - we can then reference those in the nunjucks views as if they were variables setup as view parameters by the controller
IE. this approach saves having to do this same code in each and every controller method